internal/services/datadog/datadog_monitors_resource.go

```diff
@@ -0,0 +1,493 @@
+	if d.Get("monitoring_status").(bool) {
+		monitoringStatus = datadog.MonitoringStatusEnabled
+	}
+
+	body := datadog.MonitorResource{
+		Location: utils.String(location.Normalize(d.Get("location").(string))),
+		Identity: expandMonitorIdentityProperties(d.Get("identity").([]interface{})),
+		Sku:      expandMonitorResourceSku(d.Get("sku").([]interface{})),
+		Properties: &datadog.MonitorProperties{
+			DatadogOrganizationProperties: expandMonitorOrganizationProperties(d.Get("datadog_organization_properties").([]interface{})),
+			UserInfo:                      expandMonitorUserInfo(d.Get("user_info").([]interface{})),
+			MonitoringStatus:              monitoringStatus,
+		},
+		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+	future, err := client.Create(ctx, resourceGroup, name, &body)
+	if err != nil {
+		return fmt.Errorf("creating Datadog Monitor %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for creation of the Datadog Monitor %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.SetId(id)
+	return resourceDatadogMonitorRead(d, meta)
+}
+
+func resourceDatadogMonitorRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Datadog.MonitorsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.DatadogMonitorID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.MonitorName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Datadog monitor %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving Datadog Monitor %q (Resource Group %q): %+v", id.MonitorName, id.ResourceGroup, err)
+	}
+	d.Set("name", id.MonitorName)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("location", location.NormalizeNilable(resp.Location))
+	if err := d.Set("identity", flattenMonitorIdentityProperties(resp.Identity)); err != nil {
+		return fmt.Errorf("setting `identity`: %+v", err)
+	}
+	if props := resp.Properties; props != nil {
+		if err := d.Set("datadog_organization_properties", flattenMonitorOrganizationProperties(props.DatadogOrganizationProperties, d)); err != nil {
+			return fmt.Errorf("setting `datadog_organization_properties`: %+v", err)
+		}
+		if err := d.Set("user_info", flattenMonitorUserInfo(props.UserInfo, d)); err != nil {
+			return fmt.Errorf("setting `user_info`: %+v", err)
+		}
+		d.Set("monitoring_status", props.MonitoringStatus == datadog.MonitoringStatusEnabled)
+		d.Set("liftr_resource_category", props.LiftrResourceCategory)
+		d.Set("liftr_resource_preference", props.LiftrResourcePreference)
+		d.Set("marketplace_subscription_status", props.MarketplaceSubscriptionStatus)
+	}
+	d.Set("sku", flattenMonitorResourceSku("Linked"))
```

Comment:
The input for flattenMonitorResourceSku is always "Linked"?

Reply:
Yes
Currently we are only "Linked" sku type

Reply:
Got it, but I think passing `Sku` from response makes more sense, in case there'll be more sku types?

Reply:
Updated it.